liblwgeom/lwgeom_geos.c

```diff
@@ -2118,4 +2118,31 @@ lwgeom_concavehull(const LWGEOM* geom, double ratio, uint32_t allow_holes)
 	GEOS_FREE(g1, g3);
 	return result;
 }
+
+LWGEOM*
+lwgeom_simplify_polygonal(const LWGEOM* geom, double vertex_fraction, uint32_t is_outer)
+{
+	LWGEOM* result;
+	int32_t srid = RESULT_SRID(geom);
+	uint8_t is3d = FLAGS_GET_Z(geom->flags);
+	GEOSGeometry *g1, *g3;
+
+	initGEOS(lwnotice, lwgeom_geos_error);
+
+	if (!(g1 = LWGEOM2GEOS(geom, AUTOFIX))) GEOS_FAIL();
+
+	g3 = GEOSPolygonHullSimplify(g1, is_outer, vertex_fraction);
```

Comment:
I think word "hull" is more important here than "polygon".

What happens if you pass lines or multipoints? Can you chain this a-la ST_RemoveRepeatedPoints / ST_Simplify to only affect polygons on input?

Reply:
Right now we pass stuff down to GEOS which eventually complains about non-polygonal input. Could execute a passthrough at the PostGIS level or at at the GEOS level (probably the PostGIS level).